USEPA/mywaterway
# Feature/182 bad search spinner

## Related Issues:
* [HMW-182](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=864&view=detail&selectedIssue=HMW-182)

## Main Changes:
* Updated conditional logic to differentiate lack of data from unresolved fetching. This caused an infinite loading spinner on the Overview tab in cases of bad searches.
* Fixed an attribute error on the Monitoring and Protect tabs that occurred when there is no data.
* Added a state update to clear previously populated data, which wasn't occurring for a bad search.

## Steps To Test:
1. Go to the Community page
2. Enter in an address for a valid watershed, click "Go"
3. After it loads, enter gibberish into the search bar, click "Go"
4. Check that the page resolves with the correct error messages

## TODO:
- [ ] Still have issues on the Protect tab, primarily when moving from a valid search to an invalid one without a page reload
